configurationreader/reader_test.go

```diff
@@ -21,6 +22,8 @@ var expectedConfig = Config{
 
 type ParameterStoreClientMock struct{}
 
+type ParameterStoreUnauthorizedClientMock struct{}
```

Comment:
Hvis du ikke er klar over det så kan du bruke mock funksjonaliteten til testify biblioteket her.

Da vil den opprinnelige structene endres til å bli slik:
```

type ParameterStoreClientMock struct {
   mock.Mock
}

func (ParameterStoreClientMock) GetParameter(ctx context.Context, params *ssm.GetParameterInput, optFns ...func(*ssm.Options)) (*ssm.GetParameterOutput, error) {
   args := m.Called(number)
  return args.Get(0).(*ssm.GetParameterOutput), args.Error(1)
}

```

Også kan du i unit testen i stedt skrive:

`mock.On("GetParameter", ctx, mock.Anything, mock.Anything).Return(nil, errors.New("operation error SSM: GetParameter, failed to sign request: failed to retrieve credentials")`

Reply:
Kjente ikke til det, takk! Burde være nyttig for http-kall osv. også 👍